packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```diff
@@ -231,7 +231,9 @@ function ChallengeSubmissionStepper({
       onCancel={hide}
       onStepChange={handleStepChange as (arg: string) => void}
       open={isShowingModal}
-      onConfirm={onConfirmHandler}
+      onConfirm={() => {
+        onConfirmHandler()
+      }}
```

Comment:
Updated the types , ultimately `onConfirm` cannot be a promise, because it is passed to a button's `onClick`. Call the promise in a non-promise-returning function.